doc: add component and ui element docs (#102)
Terasology/Health: docs/ui-damage-overlay.md

# Damage Overlay

The `Health` module adds a visual overlay whenever the you are damaged.
If you hurt yourself, for instance through drowning or fall damage, the default damage overlay is displayed.
It consists of four red indicators, one on each side of your screen.

> The default damage overlay is also shown when you are hurt by an effect such as poison.

Reviewer: See comment above. Here we write about "hurt by an effect such as poison" as opposed to "hurt yourself", but in the Damage section we list "poison" as example for self-inflicted damage...

Reviewer 2: This might again warrant a bit more distinctiveness (or a strive toward that at some point in the future)

* Player X died from fall damage after jumping off a cliff!
* Player X died from fall damage after being pushed off a cliff by <instigator>
* Player X died from poisoning after eating some bad puffer fish!
* Player X died from poisoning after being hit by a grave-seeker arrow shot by <instigator>!

Actually, those examples are less about hurting oneself as being about whether there is another instigator or not 🤔

Author: see my comment above

Reviewer: Maybe like this? We directly pick up the same wording as used in another place already.

> Self-inflicted causes include, for instance, falling, drowning, or poison damage.
> Damage inflicted by a third party typically involves some form of hitting the affected entity.

```suggestion
If you hurt yourself (self-inflicted damage), for instance through falling, drowning, or poisoning, the default damage overlay is displayed.
It consists of four red indicators, one on each side of your screen.
```
